src/task.c

```diff
@@ -0,0 +1,32 @@
+#include <stdint.h>
+
+#include "task.h"
+#include "asm.h"
+#include "malloc.h"
+
+static unsigned int *tasks[TASK_LIMIT];
+
+void    task_add(void (*task_code)(void), uint16_t stack_size)
+{
+    int i = 0;
+
+    while (tasks[i])
+        i++;
+    tasks[i] = (unsigned int*)malloc(sizeof(unsigned int) * stack_size);
```

Comment:
ouch. I expected `stack_size` to be size of stack in bytes, not in number of integers (which can be different on different platforms)

Reply:
Here I followed an article. I didn't really get why there int was used, my only guess is that it's convenient to calculate space used for registers and store pointer to task code. Should I change it?